src/loki_config.py

```diff
@@ -57,17 +57,12 @@
 ACTIVE_INDEX_DIR = os.path.join(LOKI_DIR, "index")
 CACHE_DIR = os.path.join(LOKI_DIR, "index_cache")
 
-
-class LokiRolesConfig:
-    """Define the configuration for Loki roles.
-
-    This object implements the ClusterRolesConfig Protocol.
-    """
-
-    roles: Iterable[str] = ROLES
-    meta_roles: Mapping[str, Iterable[str]] = META_ROLES
-    minimal_deployment: Iterable[str] = MINIMAL_DEPLOYMENT
-    recommended_deployment: Dict[str, int] = RECOMMENDED_DEPLOYMENT
+LOKI_ROLES_CONFIG = ClusterRolesConfig(
+    roles=ROLES,
+    meta_roles=META_ROLES,
+    minimal_deployment=MINIMAL_DEPLOYMENT,
+    recommended_deployment=RECOMMENDED_DEPLOYMENT,
+)
```

Comment:
Why do we need the `ROLES` etc. variables if we can assign them directly here?
Is is just to avoid the code duplication saved by `"all": set(ROLES)`?
BTW, is it correct that the `all` metaroles includes itself?

Reply:
I checked in the folder and there are no usages of those variables elsewhere in Loki except for `"all": set(ROLES)`. From my understanding these can be assigned directly as you suggested and according to our code, the all meta role must include itself because the coherence checker looks to see if the META role values are a subset of the ROLES.

Note: I do not know the purpose of this but that is what the code says.